### PR TITLE
#947 ServoMonitorCache and GC Overhead Limit Reached: Added documentation/heads up accordingly

### PR DESCRIPTION
**Pull request by tiny-dancer**
Quickly put this together - wasn't able to render a local "preview".  Feel free to criticize verbiage or syntax accordingly.

https://github.com/spring-cloud/spring-cloud-netflix/issues/947#issuecomment-247816628

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@tiny-dancer Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1370)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1370) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@tiny-dancer Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1370)!

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 1]
```diff
@@ -1728,6 +1728,20 @@ If Spring AOP is enabled and `org.aspectj:aspectjweaver` is present on your runt
 3. URI, sanitized for Atlas
 4. Client name
 
+[TIP]
+====
+Don't use hardcoded url parameters within `RestTemplate'.  When targeting dynamic endpoints use templated urls. This will avoid potential "GC Overhead Limit Reached" issues where `ServoMonitorCache` treats each url as a unique key.
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]: - s/Don't use/Avoid using
- s/use templated urls/[use URL variables](http://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html)

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 2]
```diff
@@ -1728,6 +1728,20 @@ If Spring AOP is enabled and `org.aspectj:aspectjweaver` is present on your runt
 3. URI, sanitized for Atlas
 4. Client name
 
+[TIP]
+====
+Don't use hardcoded url parameters within `RestTemplate'.  When targeting dynamic endpoints use templated urls. This will avoid potential "GC Overhead Limit Reached" issues where `ServoMonitorCache` treats each url as a unique key. 
+`
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]:
```
[source,java,indent=0]
----
```

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 3]
```diff
@@ -1728,6 +1728,20 @@ If Spring AOP is enabled and `org.aspectj:aspectjweaver` is present on your runt
...
+// recommended
+String orderid = "1";
+restTemplate.getForObject("http://testeurekabrixtonclient/orders/{orderid}", String.class, orderid)
+`
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]: remove

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 4]
```diff
@@ -1728,6 +1728,20 @@ If Spring AOP is enabled and `org.aspectj:aspectjweaver` is present on your runt
...
+`
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]: remove

File: docs/src/main/asciidoc/spring-cloud-netflix.adoc [hunk 5]
```diff
@@ -1728,6 +1728,20 @@ If Spring AOP is enabled and `org.aspectj:aspectjweaver` is present on your runt
...
+// avoid
+restTemplate.getForObject("http://testeurekabrixtonclient/orders/1", String.class)
+`
```
Comment (spencergibb) [inline docs/src/main/asciidoc/spring-cloud-netflix.adoc]:
```
----
```

**Comment by tiny-dancer** [comment]
@spencergibb believe updates are in place accordingly

**Comment by spencergibb** [comment]
Squashed and merged with some polish via d23aa67f7b144010c530cb39f43dfa8423c8560c

**Pull request closed without merging**